src/events/config.go

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright © 2020, Oracle and/or its affiliates. All rights reserved.
+ * The Universal Permissive License (UPL), Version 1.0
+ */
+package events
+
+import (
+	"os"
+
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/streaming"
+)
+
+// EnvironmentConfigurationProvider uses environment variables to get OCI config
+func EnvironmentConfigurationProvider() (common.ConfigurationProvider, error) {
+	pass := os.Getenv("PASSPHRASE")
+	provider := common.NewRawConfigurationProvider(
+		os.Getenv("TENANCY"),
+		os.Getenv("USER_ID"),
+		os.Getenv("REGION"),
+		os.Getenv("FINGERPRINT"),
+		os.Getenv("PRIVATE_KEY"),
+		&pass,
+	)
+	_, err := common.IsConfigurationProviderValid(provider)
+
+	return provider, err
+}
+
+// GetStreamClient returns a streaming client with the given configuration provider
+func GetStreamClient(provider common.ConfigurationProvider) (streaming.StreamClient, error) {
+	var endpoint string
+	endpoint = os.Getenv("MESSAGES_ENDPOINT")
```

Comment:
could this be just `endpoint := os.Getenv("MESSAGES_ENDPOINT")` instead of explicitly declaring it on line 32?

Reply:
It could, but then it would always require 2️⃣ properties, instead of just the 1️⃣  `STREAM_ID`. Essentially it's a fallback logic in cases where OSB is not used, and a simpler `oss-connection` secret definition is desired.